docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,190 @@
+[source, java]
+----
+@Entity
+public class Employee {
+
+  @Id
+  @GeneratedValue
+  public Long id;
+}
+----
+
+This results in sequential IDs that are https://cloud.google.com/spanner/docs/schema-and-data-model#primary_keys[not optimal for Cloud Spanner] and requires locking of the `hibernate_sequence` table for inserts.
+
+You should use locally-generated UUID key values instead:
+[source, java]
+----
+@Entity
+public class Employee {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  @Type(type="uuid-char")
+  public UUID id;
+}
+----
+
+The `@Type(type="uuid-char")` annotation specifies that this UUID value will be stored in Cloud Spanner as a `STRING` column.
+Leaving out this annotation causes a `BYTES` column to be used.
+
+
+==== Table Catalogs and Schemas
+
+Cloud Spanner does not support table names with Catalog and Schema components:
+
+[source, java]
+----
+// Not supported: `public.store.book` is not a valid Cloud Spanner table name.
+@Table(
+  catalog = "public",
+  schema = "store",
+  name = "book"
+)
+
+// Supported.
+@Table(
+  name = "book"
+)
+----
+
+
+==== Column Default Values
+
+The dialect does not currently set default values based on the https://docs.jboss.org/hibernate/orm/5.4/javadocs/org/hibernate/annotations/ColumnDefault.html[`@ColumnDefault`] annotation,
+because `null` values aren't specially handled and are stored just like other values by Cloud Spanner and its driver.
+
+
+==== Decimal and Numeric Types
+
+Cloud Spanner does not provide native support for https://cloud.google.com/spanner/docs/storing-numeric-data[arbitrary-precision decimal numbers], but does provide some alternative approaches.
+As a result, the driver and dialect do not support decimal and arbitrary-precision Java types that Hibernate maps to Java Persistence types `java.sql.Type` of `NUMERIC` and `DECIMAL` such as `java.math.BigDecimal`.
+
+
+==== Subclasses using `InheritanceType.JOINED`
+
+If you are using entities that are related by inheritance with the `@Inheritance(strategy = InheritanceType.JOINED)`:
+
+[source, java]
+----
+@Entity
+@Inheritance(strategy = InheritanceType.JOINED)
+public class Payment {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    private Long amount;
+}
+
+@Entity
+public class WireTransferPayment extends Payment {
+}
+
+@Entity
+public class CreditCardPayment extends Payment {
+}
+----
+
+You must set the `hibernate.hql.bulk_id_strategy` setting in `hibernate.properties` to one of the following classes:
+
+- `InlineIdsInClauseBulkIdStrategy`
+- `InlineIdsSubSelectValueListBulkIdStrategy`
+- `InlineIdsOrClauseBulkIdStrategy`
+- `CteValuesListBulkIdStrategy`
+
+This is because Hibernate's default behavior (`PersistentTableBulkIdStrategy`) attempts to create intermediate tables to handle delete and update operations on the multiple tables that represent a `JOINED` inheritance hierarchy, but these table creations statements do not conform to Cloud Spanner DDL.
```

Comment:
BTW, can we define our own strategy and use it? Like a modified `PersistentTableBulkIdStrategy` that actually works?

Reply:
For what purpose? The 4 alternatives cover the bases pretty well.

Reply:
Ooh do you mean basically just modify it so it uses the our DDL -generating-code ? yea we could look into that.

Reply:
#90